File: ETicaretAPI/Core/ETicaretAPI.Domain/Entities/Order.cs
using ETicaretAPI.Domain.Entities.Common;
using ETicaretAPI.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ETicaretAPI.Domain.Entities
{
    public class Order : BaseEntities
    {
        public Guid CustomerId { get; set; }
        public Guid AddressId { get; set; }
        public string Description { get; set; }
        public decimal TotalPrice { get; set; } 
        public OrderStatus OrderStatus { get; set; }
        public ICollection<Product> Products { get; set; }
        public Customer Customer { get; set; }
        public Address Address { get; set; }

    }
}

File: ETicaretClient/src/app/contracts/order.ts
export enum OrderStatus {
    Pending = 'Pending',
    Confirmed = 'Confirmed',
    Shipped = 'Shipped',
    Delivered = 'Delivered',
    Cancelled = 'Cancelled'
}

export class Order {
    id: string;
    customerId: string;
    addressId: string;
    description: string;
    totalPrice: number;
    orderStatus: OrderStatus;
    createDate: Date;
}